File: backend/seed.py
from models import Merchant, Discount
import random
import string

# Seed with 5 merchants and 2 discounts each
Discount.delete().execute()
Merchant.delete().execute()
for i in range(5):
    merchant = Merchant(name=f"Merchant {i}", type=random.randint(1, 2))
    merchant.save()

    for j in range(1, 3):
        discount = Discount.create(merchant=merchant, title=f"Title {j} of Merchant {i}", code="".join(
            [random.choice(string.ascii_letters) for _ in range(7)]), description="some description")
        discount.save()

File: backend2/src/utils/seed.ts
import { readFileSync } from "fs";
import { Session } from "../models/session";
import model from "../models/associations";
const { Discount, Merchant, Subscription, User } = model;

const models = [Subscription, Discount, Merchant, User, Session];

models.forEach((model) => model.destroy({ where: {}, force: true }));

function getRandomInt(min: number, max: number) {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min) + min); // The maximum is exclusive and the minimum is inclusive
}

const data = JSON.parse(
  readFileSync("src/utils/data.json", { encoding: "utf8" })
) as Record<string, string>;

(async () => {
  for (const [merchantName, image] of Object.entries(data)) {
    const merchant = await Merchant.create({
      name: merchantName,
      type: getRandomInt(1, 2),
      image,
    });

    for (let i = 1; i < 3; i++) {
      const discount = await Discount.create({
        title: `Title ${i} of merchant`,
        description: "some description",
        code: [...Array(7)].map(() => Math.random().toString(36)[2]).join(""),
      });
      await merchant.addDiscount(discount);
    }
  }
})();
